### PR TITLE
add authority levels

### PR DESCRIPTION
**Pull request by cap10morgan**
For pairing users up with their voter reg. and election authorities. Only associated with states for now.

One part of [this card](https://www.pivotaltracker.com/story/show/105559608). It calls for displaying the reg. authority's mailing address on the done step. So first we need to know which OCD Division level to use to match the user to their reg. authority, and there may or may not be an upcoming election (but you should still be able to register to vote). Hence, pulling this from election-authority-works. It's in the data model, but we hadn't run into a need for it to be implemented until now.

### CODE REVIEW

**Comment by cap10morgan** [comment]
@tank157 No need to review this yet (unless you really want to). I need to add import/export support for these values. That's coming in another commit.

**Comment by cap10morgan** [comment]
@tank157 OK those are in place now.

**Comment by cap10morgan** [comment]
Oops, I'll clean up that test failure first. :)

**Comment by tank157** [comment]
Can you explain what this ocd-child(ren) concept is here? It's not really very clear from the code, and I don't think it appears to be children in the parent/child heirarchical nature of OCD-IDs? It's just very unclear without knowing what the data structures look like what any of those are really doing.

**Comment by cap10morgan** [comment]
@tank157 It's a generalization of what used to just be a map of "methods" in that 3rd spot in the ocd-division vector. It can now include things that aren't methods, so a more generic term was needed.

If I had my druthers, we'd drop the vector representation altogether and just use a map for the whole thing. But that's a bigger change.

`["ocd-division/country:us/state:co" "Colorado" {children-map}]` <- `children-map` in there.

**Comment by tank157** [comment]
Ah, so it's essentially like a properties hash now. Some of the things in there are voting-methods or reg-methods, and some of the things in there are authority levels. I agree, we're straining the vector form here, can we put a technical debt card in to address this later with either a hash or perhaps a defrecord or something of that ilk?

File: src/election_authority_works/handlers.clj [hunk 1]
```diff
@@ -88,8 +88,12 @@
                    {(s/optional-key :voter-registration-methods)
                     #{voter-registration-method-schema}
                     (s/optional-key :voting-methods)
-                    #{voting-method-schema}}
-                   "Voter reg. and/or voting methods")]})
+                    #{voting-method-schema}
+                    (s/optional-key :voter-registration-authority-level)
+                    (s/enum :state :county :municipal)
+                    (s/optional-key :election-authority-level)
+                    (s/enum :state :county :municipal)}
```
Comment (tank157) [inline src/election_authority_works/handlers.clj]: (Very minor suggestion) We could just def the authority-level enums once and plug it into both spots here.

File: src/election_authority_works/entities/authority.clj [hunk 2]
```diff
@@ -149,19 +160,24 @@
...
         (tx/dissoc-optional-nil-values
          {:voter-registration-methods
           (not-empty (set (map entity->voter-registration-method
                                voter-registration-method)))
           :voting-methods
           (not-empty (set (map entity->voting-method
-                               voting-method)))}
-         [:voter-registration-methods :voting-methods])]
+                               voting-method)))
+          :voter-registration-authority-level voter-registration-authority-level
```
Comment (tank157) [inline src/election_authority_works/entities/authority.clj]: Where do `voter-registration-authority-level` and `election-authority-level` come from here? Shouldn't they be namespaced?

Comment (tank157) [inline src/election_authority_works/entities/authority.clj]: I guess not, when you use the namespace in the destructuring, it strips it from the var. Weird, did not know that.

**Comment by tank157** [comment]
Ok, LGTM (with optional schema cleanup)

**Comment by cap10morgan** [comment]
Thanks, @tank157. I created [the requested card](https://www.pivotaltracker.com/story/show/109672026).

**Comment by cap10morgan** [comment]
@tank157 2 more commits to go with the new http-api gateway. They make this component work the way bifrost expects.

I've issued PRs for other components that needed to be changed b/c of these changes (assigned to other reviewers), but they looked like they could cope with both the old and new formats so no versioning necessary.

**Comment by tank157** [comment]
@cap10morgan new commits LGTM

**Pull request merged**